purchase_stock_bl_date/readme/DESCRIPTION.rst

```diff
@@ -0,0 +1,15 @@
+This modules does the following:
+
+* Add bl_date in purchase order.
+* Pass bl_date to receipts
+
+Background
+
+In an operation where the purchase order is created at the same time as the receipt of goods, which is done due to
+frequent discrepancies between the received quantity of goods and the quantity at the time of receipt, there is a
+condition where the currency rate in effect at the time of creating the purchase order differs from the currency rate
+used when the B/L (Bill of Lading) is issued. This becomes a problem when the user wants the price calculated in the
+journal entry to use the currency rate used on the B/L issuance date.
+
+This module is designed to address this issue by calculating the price in the journal entry based on the currency rate
+in effect on the specified "B/L Date" in purchase order.
```

Comment:
Here is my version:

```suggestion
This module adds B/L Date fields to the purchase order and the picking.  B/L Date assigned in the purchase receipt is used for capturing the currency rate for the unit price calculation, which is the basis of inventory valuation.

Note that this module only adjusts the logic where currency rate is captured.  Adjusting the date of stock moves or journal entries is outside the scope of this module.

Background:
~~~~~~~~~~~

Depending on the incoterms of the purchase transaction in international trades, the ownership transfer of the purchased products takes place before they arrive at the demestic warehouse (i.e. when products are loaded in cargo in the origin country), while the receipt operation in Odoo, for practical reasons, have to be done when the products are received at the warehouse.  This results in an inaccurate valuation of the inventory, which should be avoided.
```

Does this make sense?

Please also add USAGE.rst.

Reply:
>Please check if the recovery can be done under the current design in case user makes mistake with the 
>operation (e.g. forget to input B/L date).
>If the recovery operation is tricky (which I suppose it is), please suggest and adjust the desgin with Lin/Kanda-san.

Hi @yostashiro san, as you mentioned that if user forgot to set the bl date, the recovery is tricky with current design.

@AungKoKoLin1997 
Im thinking following logic, is it possible to be implemented?
- set the bl_date as read only when picking state is done (currently bl date is editable even state is done)
- add new boolean field "force bl date" after bl_date, editable only if user click unlock button on picking
- if "force bl date" is True, bl_date is editable
- if bl_date updated on picking in Done state, and user click lock button, its trigger following process:
  1. revaluate the stock valuation with the updated bl_date
  2. reverse the current related svl journal entry, and generate new journal entry with the updated information. (or if possible to directly update the current related svl journal entry is better)

Kindly your advise